cmd/root.go

```diff
@@ -438,7 +478,159 @@ func createIssue(issue github.Issue, client jira.Client) error {
 		return err
 	}
 
-	log.Debugf("Created JIRA issue #%s!", jIssue.ID)
+	log.Debugf("Created JIRA issue %s!", jIssue.Key)
+
+	if err = createComments(issue, *jIssue, nil, ghClient, jClient); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var jCommentRegex = regexp.MustCompile("^Comment \\(ID (\\d+)\\) from GitHub user (\\w+) \\((.+)\\)? at (.+):\\n\\n(.+)$")
+var jCommentIDRegex = regexp.MustCompile("^Comment \\(ID (\\d+)\\)")
+
+func createComments(ghIssue github.Issue, jIssue jira.Issue, existing []jira.Comment, ghClient github.Client, jClient jira.Client) error {
+	if *ghIssue.Comments == 0 {
+		log.Debugf("Issue #%d has no comments, skipping.", *ghIssue.Number)
+		return nil
+	}
+
+	ctx := context.Background()
+	repo := strings.Split(rootCmdCfg.GetString("repo-name"), "/")
+	comments, _, err := ghClient.Issues.ListComments(ctx, repo[0], repo[1], *ghIssue.Number, &github.IssueListCommentsOptions{
+		Sort:      "created",
+		Direction: "asc",
+	})
+	if err != nil {
+		log.Errorf("Error retrieving GitHub comments for issue #%d. Error: %s.", *ghIssue.Number, err)
+		return err
+	}
+
+	for _, ghComment := range comments {
+		found := false
+		for _, jComment := range existing {
+			if !jCommentIDRegex.MatchString(jComment.Body) {
+				continue
+			}
+			// matches[0] is the whole string, matches[1] is the ID
+			matches := jCommentIDRegex.FindStringSubmatch(jComment.Body)
+			id, _ := strconv.Atoi(matches[1])
+			if *ghComment.ID != id {
+				continue
+			}
+			found = true
+
+			updateComment(*ghComment, jComment, jIssue, ghClient, jClient)
+		}
+		if found {
+			continue
+		}
+
+		if err := createComment(*ghComment, jIssue, ghClient, jClient); err != nil {
+			return err
+		}
+	}
+
+	log.Debugf("Copied comments from GH issue #%d to JIRA issue %s.", *ghIssue.Number, jIssue.Key)
+	return nil
+}
+
+func updateComment(ghComment github.IssueComment, jComment jira.Comment, jIssue jira.Issue, ghClient github.Client, jClient jira.Client) error {
+	// fields[0] is the whole body, 1 is the ID, 2 is the username, 3 is the real name (or "" if none)
+	// 4 is the date, and 5 is the real body
+	fields := jCommentRegex.FindStringSubmatch(jComment.Body)
+
+	if fields[5] == *ghComment.Body {
+		return nil
+	}
+
+	user, _, err := ghClient.Users.Get(context.Background(), *ghComment.User.Login)
+	if err != nil {
+		log.Errorf("Error retrieving GitHub user %s. Error: %s", *ghComment.User.Login, err)
+	}
+	body := fmt.Sprintf("Comment (ID %d) from GitHub user %s", *ghComment.ID, user.GetLogin())
+	if user.GetName() != "" {
```

Comment:
To simplify this a little bit, how about making the only bit that is conditioned on `GetName() != ""` be the part that actually appends the username:
```go
if user.GetName() != "" {
    body - fmt.Sprintf("%s (%s)", body, user.GetName())
}
body = fmt.Sprintf(
    "%s at %s:\n\n%s",
    body,
    ghComment.CreatedAt.Format(commentDateFormat),
    *ghComment.Body,
)
```
This cleans up the logic, de-dupes the code and shortens everything a little.